src/test/java/com/rtb/projectmanagementtool/task/TaskTest.java

```diff
@@ -0,0 +1,259 @@
+    Assert.assertEquals("status", status1, task.getStatus());
+    Assert.assertEquals("users", users1, task.getUsers());
+    Assert.assertEquals("subtasks", subtasks1, task.getSubtasks());
+  }
+
+  @Test
+  public void testCreateTaskFromEntity() {
+    // Build entity
+    Entity entity = new Entity("Task", taskID2);
+    entity.setProperty("projectID", projectID2);
+    entity.setProperty("name", name2);
+    entity.setProperty("description", description2);
+    entity.setProperty("status", status2.name());
+    entity.setProperty("users", users2);
+    entity.setProperty("subtasks", subtasks2);
+
+    // Build TaskData object from entity
+    TaskData task = new TaskData(entity);
+
+    // Assert TaskData parameters were stored correctly
+    Assert.assertEquals("taskID", taskID2, task.getTaskID());
+    Assert.assertEquals("projectID", projectID2, task.getProjectID());
+    Assert.assertEquals("name", name2, task.getName());
+    Assert.assertEquals("description", description2, task.getDescription());
+    Assert.assertEquals("status", status2, task.getStatus());
+    Assert.assertEquals("users", users2, task.getUsers());
+    Assert.assertEquals("subtasks", subtasks2, task.getSubtasks());
+  }
+
+  @Test
+  public void testCreateEntityFromTask() {
+    // Build TaskData object
+    TaskData task =
+        new TaskData(taskID3, projectID3, name3, description3, status3, users3, subtasks3);
+
+    // Create task entity from TaskData object
+    Entity entity = task.toEntity();
+
+    // Get task entity attributes
+    long entityTaskID = (long) entity.getKey().getId();
+    long entityProjectID = (long) entity.getProperty("projectID");
+    String entityName = (String) entity.getProperty("name");
+    String entityDescription = (String) entity.getProperty("description");
+    Status entityStatus = Status.valueOf((String) entity.getProperty("status"));
+    HashSet<Long> entityUsers = (HashSet<Long>) entity.getProperty("users");
+    HashSet<Long> entitySubtasks = (HashSet<Long>) entity.getProperty("subtasks");
+
+    // Assert task entity attributes equal TaskData attributes
+    Assert.assertEquals("taskID", taskID3, entityTaskID);
+    Assert.assertEquals("projectID", projectID3, entityProjectID);
+    Assert.assertEquals("name", name3, entityName);
+    Assert.assertEquals("description", description3, entityDescription);
+    Assert.assertEquals("status", status3, entityStatus);
+    Assert.assertEquals("users", users3, entityUsers);
+    Assert.assertEquals("subtasks", subtasks3, entitySubtasks);
+  }
+
+  @Test
+  public void testGetTasksFromEmptyDs() {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+    // Assert no task entities are found
+    Assert.assertEquals(0, ds.prepare(new Query("Task")).countEntities(withLimit(10)));
+
+    // Get task entities with TaskController
+    TaskController taskController = new TaskController();
```

Comment:
I think this helps illustrate why moving the Datastore out of TaskController could be good.  The way you have it here right now, the test in line 179 is using a different instance of Datastore than the one that's inside of TaskController.  In this instance they both connect to the same database (since the API takes care of that detail), but if you were using a different database engine, it could be possible for the one DS to connect to an entirely different database than the other.  (A common example would be where one connects to a Test database and the other connects to a Prod database).